File: app/src/main/java/io/eodc/planit/db/SubjectDao.java
package io.eodc.planit.db;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

@Dao
public interface SubjectDao {
    @Query("SELECT * FROM subjects")
    List<Subject> getAllSubjects();

    @Query("SELECT * FROM subjects")
    LiveData<List<Subject>> getAllSubjectsObservable();

    @Query("SELECT * FROM subjects WHERE id = :id")
    Subject getClassById(int id);

    @Update
    void updateSubjects(Subject... updatedSubjects);

    @Insert
    void insertSubjects(Subject... insertedSubjects);

    @Delete
    void removeSubjects(Subject... removedSubjects);
}

File: app/src/main/java/io/eodc/planit/db/SubjectDao.kt
package io.eodc.planit.db

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface SubjectDao {
    @get:Query("SELECT * FROM subjects")
    val allSubjects: List<Subject>

    @get:Query("SELECT * FROM subjects")
    val allSubjectsObservable: LiveData<List<Subject>>

    @Query("SELECT * FROM subjects WHERE id = :id")
    fun getClassById(id: Int): Subject

    @Update
    fun updateSubjects(vararg updatedSubjects: Subject)

    @Insert
    fun insertSubjects(vararg insertedSubjects: Subject)

    @Delete
    fun removeSubjects(vararg removedSubjects: Subject)
}
